roles/ora-host/tasks/hugepages.yml

```
@@ -18,14 +18,29 @@
   register: v_Hugepagesize
   changed_when: false
 
+- name: hugepages | Calculate number of hugepages required
+  set_fact:
+    v_hugepages_required:  "{{ ((ora_sga_target_mb | int) * 1024 / (v_Hugepagesize.stdout | int) * (hugepages_ratio | float)) | round(0,'ceil') | int }}"
+
 - name: hugepages | Update the vm.nr_hugepages sysctl value
   sysctl:
     name: "vm.nr_hugepages"
-    value: "{{ ((ora_sga_target_mb | int) * 1024 / (v_Hugepagesize.stdout | int) * (hugepages_ratio | float)) | round(0,'ceil') | int }}"
+    value: "{{ v_hugepages_required | string }}"
     state: present
     sysctl_set: true
     reload: true
 
+- name: hugepages | Fetch actual value of vm.nr_hugepages
+  command: sysctl -n vm.nr_hugepages
+  register: v_actual_hugepages_allocated
+  changed_when: false
+
+- name: hugepages | Assert that vm.nr_hugepages matches the expected value of {{ v_hugepages_required }}
```

So from what I see, we first set vm.nr_hugepages, and only then verify that the actual value matches the expected one.

I think it might be more cleaner to check available memory (grep MemAvailable /proc/meminfo) before attempting to set vm.nr_hugepages and failing early. Although It's a minor detail and doesn't impact functionality, so I'm not blocking this. LGTM

---

Fair point.  Now that we have a prerequisite check playbook, I'll create a PR to try and add an earlier validation too.